kik_unofficial/client.py

```diff
@@ -320,10 +321,17 @@ def request_info_of_users(self, peer_jids: Union[str, List[str]]):
                           If you want to request information for more than one user, supply a list of strings.
                           Otherwise, supply a string
         """
+
         return self._send_xmpp_element(roster.QueryUsersInfoRequest(peer_jids))
 
     def add_friend(self, peer_jid):
-        return self._send_xmpp_element(roster.AddFriendRequest(peer_jid))
+        message = roster.AddFriendRequest(peer_jid)
+        if "_a@talk" in peer_jid:
+            self.alias_jid_cache[message.message_id] = peer_jid
```

Comment:
So `alias_jid_cache` is more of a mapping between sent friend requests and their associated JIDs. Maybe we can save the whole `AddFriendRequest`s as an array or make a better name for this variable.

Reply:
I changed the name of this mapping to friend_request_mapping. 

Its only purpose is to hold the message_id of a friend request. So that when the friend request is returned we can add the alias_jid it was requested on to the Peer Info Response message. It is deleted once it is matched to a response.